src/managementgroups/azext_managementgroups/custom.py

```diff
@@ -0,0 +1,109 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core._profile import Profile
+
+
+def _register_rp(cli_ctx, subscription_id=None):
+    rp = "Microsoft.Management"
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.cli.core.profiles import ResourceType
+    import time
+    rcf = get_mgmt_service_client(
+        cli_ctx,
+        ResourceType.MGMT_RESOURCE_RESOURCES,
+        subscription_id)
+    rcf.providers.register(rp)
+    while True:
+        time.sleep(10)
+        rp_info = rcf.providers.get(rp)
+        if rp_info.registration_state == 'Registered':
+            break
+
+
+def _get_subscription_id_from_subscription(cli_ctx, subscription):
+    profile = Profile(cli_ctx=cli_ctx)
+    subscriptions_list = profile.load_cached_subscriptions()
+    for sub in subscriptions_list:
+        if sub['id'] == subscription or sub['name'] == subscription:
+            return sub['id']
+    from azure.cli.core.util import CLIError
+    raise CLIError("Subscription not found in the current context.")
+    return None
```

Comment:
The return is redundant, please remove.

Reply:
Was not present earlier but caused a static error.